projects/distributed-tracing/src/shared/dashboard/widgets/table/table-widget.model.ts

```diff
@@ -60,6 +61,27 @@ export class TableWidgetModel {
   })
   public columns: TableWidgetColumnModel[] = [];
 
+  @ModelProperty({
+    key: 'searchAttribute',
+    displayName: 'Search Attribute',
+    type: STRING_PROPERTY.type
+  })
+  public searchAttribute?: string;
+
+  @ModelProperty({
+    key: 'filterToggles',
+    displayName: 'Filter Toggles',
+    // tslint:disable-next-line: no-object-literal-type-assertion
+    type: {
+      key: ARRAY_PROPERTY.type,
+      subtype: {
+        key: ModelPropertyType.TYPE,
+        defaultModelClass: TableWidgetFilterModel
+      }
+    } as ArrayPropertyTypeInstance
+  })
+  public filterToggles: TableWidgetFilterModel[] = [];
```

Comment:
I fear we're going to quickly be overrun by different controls, but let's see what happens

Reply:
Yeah... I hear you. But there is only so much we can stuff in the header for a table. Not sure what else could be added (though I'm sure we'll find something).

I get wanting to compose bespoke headers so we don't have these uber components with a bunch of Input options that may not always make sense for each use case, but I also want to force us into a consistent experience. The single component has some pros and cons. For consistency, it makes sense to me to club all of these controls into a single component with our current use cases, but agree its a slippery slope.

Reply:
Two that come to mind are the filter bar and then various action menus. So may be worth starting to think how those might fit in.